Пятая лабораторная работа выполнена

PopovMatvey opened the pull request:
Пятая лабораторная работа была выполнена с соблюдёнными условиями:
1) Для калькулятора были разработаны функции:
add();
subtract();
divide();
multiply();
calculator.getResult();
calculator.reset();
2) Состояние результата сохраняется;
3) Функцию можно вызывать следующим образом:
calculator.subtract(1)(1)(1)(2);
4) Также функцию можно вызывать:
calculator.add()();
5) Все тесты и линтеры были пройдены.

PopovMatvey committed:
Пятая лабораторная работа выполнена

Пятая лабораторная работа была выполнена с соблюдёнными условиями:
1) Для калькулятора были разработаны функции:
add();
subtract();
divide();
multiply();
calculator.getResult();
calculator.reset();
2) Состояние результата сохраняется;
3) Функцию можно вызывать следующим образом:
calculator.subtract(1)(1)(1)(2);
4) Также функцию можно вызывать:
calculator.add()();
5) Все тесты и линтеры были пройдены.

djilua1 commented on src/ex5_js-functions/task.js:
```diff
@@ -0,0 +1,101 @@
+
+calculator.multiply = calculator.multiply();
+
+calculator.divide = calculator.divide();
```
Это зачем так?

djilua1 commented on src/ex5_js-functions/task.js:
```diff
@@ -0,0 +1,101 @@
+        result /= value;
+
+        return functionRecurs;
+      }
```
Это излишне.

djilua1 commented on src/ex5_js-functions/task.js:
```diff
@@ -0,0 +1,101 @@
+        return functionRecurs;
+      }
+
+      if ((typeof value !== 'number') || (Number.isNaN(value)) || (typeof value === 'undefined')) {
```
Достаточно обработать момент когда аргумент не передается.

djilua1 reviewed (commented):
По большинству методов идентичные замечания.

PopovMatvey committed:
Исправления по пятой лабораторной работе

Были исправлены три из трёх замечаний:
1) Убрал ненужные присвоения перед экспортом;
2) Упростил реализацию сложения, вычитания, умножения и деления;
3) Теперь я проверяю только если аргумент не передался при сложении, вычитании, умножении и делении.
Также тесты и линтеры все выполнены.
